### PR TITLE
[no-release-notes] Include indexes and cacheability in plan tests

### PR DESCRIPTION
**Pull request by max-hoffman**
companion: https://github.com/dolthub/dolt/pull/5014

### CODE REVIEW

File: sql/plan/resolved_table.go [hunk 1]
```diff
@@ -68,15 +69,15 @@ func (t *ResolvedTable) String() string {
 			filters = append(filters, f.String())
 		}
 		if len(filters) > 0 {
-			pr.WriteChildren(fmt.Sprintf("filters: %v", filters))
+			children = append(children, fmt.Sprintf("filters: %v", filters))
```
Comment (zachmu) [inline sql/plan/resolved_table.go:72]: Looks like you're never calling WriteChildren after this change

File: sql/plan/hash_lookup.go [hunk 2]
```diff
@@ -64,15 +65,23 @@ func (n *HashLookup) WithExpressions(exprs ...sql.Expression) (sql.Node, error)
 
 func (n *HashLookup) String() string {
 	pr := sql.NewTreePrinter()
-	_ = pr.WriteNode("HashLookup(child: %v, lookup: %v)", n.childProjection, n.lookupProjection)
-	_ = pr.WriteChildren(n.UnaryNode.Child.String())
+	_ = pr.WriteNode("HashLookup")
+	children := make([]string, 3)
+	children[0] = fmt.Sprintf("source: %s", n.lookupProjection)
```
Comment (zachmu) [inline sql/plan/hash_lookup.go:70]: Either the name of the field is bad or the debug output is

Think on your sins

File: enginetest/queries/queries.go [hunk 3]
```diff
@@ -10971,97 +10971,36 @@ var IndexPrefixQueries = []ScriptTest{
 					{0, "a", "a"},
 				},
 			},
-			{
```
Comment (zachmu) [inline enginetest/queries/queries.go:10974]: How confident are you that we have suitable coverage for these plans elsewhere?

**Review by zachmu — APPROVED** [review]
Seems reasonable

File: enginetest/queries/queries.go (hunk 3 above)
Comment (max-hoffman) [reply enginetest/queries/queries.go:10974]: all of these are preserved, moved into `IndexQueryPlans`

**Pull request merged**